File: IAIP.Testx/ExtensionMethods/ControlExtensions.cs
using System;
using System.ComponentModel;
using System.Reflection;
using System.Windows.Forms;

namespace IAIP.Testx.ExtensionMethods
{
    public static class ControlExtensions
    {
        private static readonly MethodInfo onValidating = typeof(Control).GetMethod("OnValidating", BindingFlags.Instance | BindingFlags.NonPublic);
        private static readonly MethodInfo onValidated = typeof(Control).GetMethod("OnValidated", BindingFlags.Instance | BindingFlags.NonPublic);

        public static bool Validate(this Control control)
        {
            CancelEventArgs e = new CancelEventArgs();

            onValidating.Invoke(control, new object[] { e });
            if (e.Cancel) return false;

            onValidated.Invoke(control, new object[] { EventArgs.Empty });
            return true;
        }
    }
}

File: IAIP.Test/ExtensionMethods/ControlExtensions.vb
Imports System.ComponentModel
Imports System.Reflection
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Public Module ControlExtensions

    <Extension>
    Public Function Validate(control As Control) As Boolean
        Dim e As New CancelEventArgs()
        Dim onValidating As MethodInfo = GetType(Control).GetMethod("OnValidating", BindingFlags.Instance Or BindingFlags.NonPublic)
        Dim onValidated As MethodInfo = GetType(Control).GetMethod("OnValidated", BindingFlags.Instance Or BindingFlags.NonPublic)

        onValidating.Invoke(control, New Object() {e})
        If e.Cancel Then
            Return False
        End If

        onValidated.Invoke(control, New Object() {EventArgs.Empty})

        Return True
    End Function

End Module
